Indicate a single group or object has been dropped on the control stream (#443)
moq-wg/moq-transport: draft-ietf-moq-transport.md

from the latest arriving message, as they are delivered in order on a single
stream.

## OBJECTS_DROPPED {#message-objects-dropped}

An endpoint sends a 'OBJECTS_DROPPED' message on the control stream when
one or more Objects from a group will not be delivered for this subscription
due to congestion or other constraints.  This does not indicate the dropped
Objects are permanantly unavailable.

~~~
OBJECTS_DROPPED Message {
  Subscribe ID (i),
  Group ID (i),
}
~~~
{: #moq-track-objects-dropped title="MOQT OBJECTS_DROPPED Message"}

* Subscribe ID: Subscription Identifier as defined in {{message-subscribe-req}}.

> * Group ID: The ID of the group the subscriber does not expect any more Objects for.

Reviewer: The description makes it sound like the subscriber sends this message.  This is sent by the publisher, so maybe express it in terms of what won't be sent, and what the subscriber can expect.  

I'm also curious how it interacts with forwarding preference=track.

Author: With forwarding preference of track, I don't think this would be needed, because if you skipped Objects, it would be very clear.

> * Group ID: The ID of the group the subscriber does not expect any more Objects for.

Author: ```suggestion
* Group ID: The ID of the group the publisher will no longer send objects for.
```

from the latest arriving message, as they are delivered in order on a single
stream.

## OBJECTS_DROPPED {#message-objects-dropped}

An endpoint sends a 'OBJECTS_DROPPED' message on the control stream when
one or more Objects from a group will not be delivered for this subscription

> due to congestion or other constraints.  This does not indicate the dropped

Reviewer 2: trying to understand what could be examples of "other constraints".

Author: Flow control is one, I'm not sure what others would be.  I can imagine other resource limits coming into effect, like memory pressure, etc, but I'm not sure.

I can be more specific, but I'm not sure it's helpful?

from the latest arriving message, as they are delivered in order on a single
stream.

## OBJECT_DROPPED {#message-object-dropped}

An endpoint sends a 'OBJECT_DROPPED' message on the control stream when
one Object will not be delivered for this subscription due to congestion or
other constraints.  This does not indicate the dropped Object is permanantly
unavailable.

~~~
OBJECT_DROPPED Message {
  Subscribe ID (i),
  Group ID (i),
  Object ID (i)
}
~~~

> {: #moq-track-objects-dropped title="MOQT OBJECT_DROPPED Message"}

Author: ```suggestion
{: #moq-track-object-dropped title="MOQT OBJECT_DROPPED Message"}
```
